app/commands/solution/search_community_solutions.rb

```diff
@@ -17,25 +17,64 @@ def initialize(exercise, page: nil, per: nil, criteria: nil)
     end
 
     def call
-      @solutions = exercise.solutions.published
+      results = client.search(index: 'solutions', body: search_body)
 
-      filter_criteria!
-      sort!
+      solution_ids = results["hits"]["hits"].map { |hit| hit["_source"]["id"] }
+      solutions = Solution.where(id: solution_ids).
+        includes(:exercise, :track).
+        order(Arel.sql("FIND_IN_SET(id, '#{solution_ids.join(',')}')")).
+        to_a
 
-      @solutions.page(page).per(per)
+      total_count = results["hits"]["total"]["value"].to_i
+      Kaminari.paginate_array(solutions, total_count: total_count).
+        page(page).per(per)
```

Comment:
Just for niceness...

```suggestion
      Kaminari.paginate_array(
        solutions, 
        total_count: results["hits"]["total"]["value"].to_i
      ).page(page).per(per)
```